pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -205,3 +208,15 @@ func validateTaskRunSpec(ctx context.Context, trs PipelineTaskRunSpec) (errs *ap
 	}
 	return errs
 }
+
+// validateResourcesRequirements() validates if only step-level or task-level resources requirements are configured
+func validateResourcesRequirements(trs PipelineTaskRunSpec) (errs *apis.FieldError) {
+	for _, taskRunStepOverride := range trs.StepOverrides {
```

Comment:
I think the TEP states that we'll consider different resource types separately; so for example you could specify step level for CPU and task-level for memory. (Open to revisiting this decision.)

Also, why use `.Size > 0` instead of a nil check?

Reply:
for ` .Size > 0`, an empty/stub resources field was populated out to fail this check when I did the local test on cluster. so using this.. a bit dummy way.  😵